src/core/geocms/qgsgeocmsconnection.h

```diff
@@ -0,0 +1,78 @@
+/***************************************************************************
+    qgsgeocmsconnection.h
+    ---------------------
+    begin                : Feb 2017
+    copyright            : (C) 2017 by Muhammad Yarjuna Rohmat, Ismail Sunni
+    email                : rohmat at kartoza dot com, ismail at kartoza dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSGEOCMSCONNECTION_H
+#define QGSGEOCMSCONNECTION_H
+
+#define SIP_NO_FILE
+
+#include "qgis_core.h"
+#include "qgsdatasourceuri.h"
+#include "qgsmaplayer.h"
+#include "qgsproject.h"
+#include "qgsgeonoderequest.h"
+
+#include <QString>
+#include <QMultiMap>
+#include <QNetworkReply>
+
+
+/** \ingroup core
+ * Base class for all GeoCMS connection.
+ * Parent/children hierarchy is not based on QObject.
+*/
+class CORE_EXPORT QgsGeoCmsConnection : public QObject
```

Comment:
@wonder-sk We added this based on a discussion with @elpaso in Essen - he was concerned that (e.g. someone implementing a similar tool for geoserver / CKAN or some other spatial CMS like environment) would re-implement a lot of boilerplate code to provide similar integration.

For my part I don't mind if we remove this for now...